### PR TITLE
(feat) gcp support

### PR DESCRIPTION
**Pull request by eoinmcafee00**

### CODE REVIEW

File: command/delegate/delegate.go [hunk 1]
```diff
@@ -123,7 +123,14 @@ func (c *delegateCommand) run(*kingpin.ParseContext) error {
 		KeyFile:             config.DefaultPoolSettings.KeyFile,
 	}
 	// process the pool file
-	pools, poolFileErr := cloudaws.ProcessPoolFile(c.awsPoolfile, &c.defaultPoolSettings)
+	//pools, poolFileErr := cloudaws.ProcessPoolFile(c.awsPoolfile, &c.defaultPoolSettings)
+	//if poolFileErr != nil {
```
Comment (eoinmcafee00) [inline command/delegate/delegate.go:127]: ignore for now, will be fixed when we update to support one pool file

File: internal/vmpool/cloudaws/pool.go [hunk 2]
```diff
@@ -74,10 +74,6 @@ func (p *awsPool) GetOS() string {
 	return p.os
 }
 
-func (p *awsPool) GetUser() string {
-	return p.user
```
Comment (eoinmcafee00) [inline internal/vmpool/cloudaws/pool.go:78]: Didn't see any evidence this was being used.

File: command/daemon/daemon.go [hunk 3]
```diff
@@ -125,7 +127,13 @@ func (c *daemonCommand) run(*kingpin.ParseContext) error {
 		CertFile:            config.DefaultPoolSettings.CertFile,
 		KeyFile:             config.DefaultPoolSettings.KeyFile,
 	}
-	pools, err := cloudaws.ProcessPoolFile(c.poolFile, &defaultPoolSettings)
+	//pools, err := cloudaws.ProcessPoolFile(c.poolFile, &defaultPoolSettings)
```
Comment (marko-gacesa) [inline command/daemon/daemon.go:130]: There is no need to comment these lines out. The `poolManager` supports multiple pools. Use something like:
```
poolsAWS, err := cloudaws.ProcessPoolFile(c.poolFile, &defaultPoolSettings)
err = poolManager.Add(poolsAWS...)

poolsGCP, err := google.ProcessPoolFile(c.googlePoolFile, &defaultPoolSettings)
err = poolManager.Add(poolsGCP...)
```

We'll refactor it later when we switch to using a single pool file solution.

File: command/delegate/delegate.go (hunk 1 above)
Comment (marko-gacesa) [reply command/delegate/delegate.go:127]: yes, this is the correct implementation for now.. see my comment in the daemon.go

File: internal/vmpool/google/config.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/drone-runners/drone-runner-aws/internal/cloudinit"
```
Comment (marko-gacesa) [inline internal/vmpool/google/config.go:9]: order the imports correctly...

File: command/delegate/delegate.go (hunk 1 above)
Comment (eoinmcafee00) [reply command/delegate/delegate.go]: I've uncommented this for now - will pick up work after merge

File: internal/vmpool/google/config.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+	"github.com/drone-runners/drone-runner-aws/oshelp"
+
+	"github.com/buildkite/yaml"
+	"github.com/drone/runner-go/logger"
+
+	"github.com/drone-runners/drone-runner-aws/internal/vmpool"
```
Comment (marko-gacesa) [inline internal/vmpool/google/config.go:16]: move this line up to the correct section.

File: internal/vmpool/google/pool.go [hunk 6]
```diff
@@ -0,0 +1,507 @@
...
+	if list.Items == nil {
+		return
+	}
+
+	for _, vm := range list.Items {
+		logr.
+			WithField("machine", vm.Name).
+			WithField("id", vm.Id).
+			WithField("status", vm.Status).
+			Traceln("-------- list machines -----------")
```
Comment (marko-gacesa) [inline internal/vmpool/google/pool.go:117]: Maybe remove logging of every instance, every time the List method is called...?

File: internal/vmpool/google/config.go (hunk 5 above)
Comment (eoinmcafee00) [reply internal/vmpool/google/config.go]: done

**Review by marko-gacesa — APPROVED** [review]

**Pull request merged**